clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferMetadata.scala

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.clio.transfer.model
+
+/**
+  * Common information for metadata in all clio indexes.
+  */
+trait TransferMetadata[M <: TransferMetadata[M]] { self: M =>
+
+  /**
+    * Notes used to track changes to metadata (primarily deletions).
+    */
+  val notes: Option[String]
+
+  /**
+    * Paths to all files that should be moved by the clio-client move
+    * command for this metadata's index.
+    */
+  def pathsToMove: Seq[String]
```

Comment:
Is there any way to use `Path` for the paths in this class? I think they should be valid `Path` objects. That way you can use `Path` operations instead of string operations.

Reply:
I think we can, and we might even be able to plug in the google nio filesystem adapter to properly parse the gs:// paths. Will investigate!

Reply:
Poked at this a bit, but I think it'll turn out to be a deep rabbit hole. I'll make a tech-debt ticket to move to path-based APIs.